pulp_rpm/app/tasks/publishing.py

```diff
@@ -132,48 +155,20 @@ def populate(self):
         )
 
         for distribution_tree in distribution_trees:
-            for content_artifact in distribution_tree.contentartifact_set.all():
-                self.published_artifacts.append(PublishedArtifact(
-                    relative_path=content_artifact.relative_path,
-                    publication=publication,
-                    content_artifact=content_artifact)
-                )
-            for addon in distribution_tree.addons.all():
-                repository_version = addon.repository.latest_version()
-                if repository_version and repository_version.content != main_content:
-                    self.sub_repos.append((addon.addon_id, repository_version.content))
-            for variant in distribution_tree.variants.all():
-                repository_version = variant.repository.latest_version()
-                if repository_version and repository_version.content != main_content:
-                    self.sub_repos.append((variant.variant_id, repository_version.content))
+            self.handle_sub_repo(distribution_tree.addons.all(), "addon_id")
+            self.handle_sub_repo(distribution_tree.variants.all(), "variant_id")
 
             treeinfo_file = create_treeinfo(distribution_tree)
             PublishedMetadata.create_from_file(
-                publication=publication,
+                publication=self.publication,
                 file=File(open(treeinfo_file.name, 'rb'))
             )
 
-        self.packages = self.get_packages(main_content)
-        self.repomdrecords = self.prepare_metadata_files(main_content)
-
-        all_packages = self.packages
         for name, content in self.sub_repos:
             os.mkdir(name)
-            sub_repo_packages = self.get_packages(content)
-            all_packages = all_packages | sub_repo_packages
-            setattr(self, f"{name}_packages", sub_repo_packages)
+            setattr(self, f"{name}_contents", content)
```

Comment:
exposing `contents` instead of `packages` this change is due to kickstarts, I will explain better on the next comments